cmd/frontend/graphqlbackend/own.go

```diff
@@ -13,7 +13,28 @@ import (
 type ListOwnershipArgs struct {
 	First   *int32
 	After   *string
-	Reasons *[]string
+	Reasons *[]OwnershipReasonType
+}
+
+type OwnershipReasonType string
+
+const (
+	CodeownersFileEntry              OwnershipReasonType = "CODEOWNERS_FILE_ENTRY"
+	RecentContributorOwnershipSignal OwnershipReasonType = "RECENT_CONTRIBUTOR_OWNERSHIP_SIGNAL"
+	RecentViewOwnershipSignal        OwnershipReasonType = "RECENT_VIEW_OWNERSHIP_SIGNAL"
+)
+
+func (args *ListOwnershipArgs) ListsReason(reason OwnershipReasonType) bool {
```

Comment:
nit: sounds a bit more explicit to me that way, but totally can be a subjective thingy
```suggestion
func (args *ListOwnershipArgs) ContainsReason(reason OwnershipReasonType) bool {
```

Reply:
Perhaps `ShouldListReason`, since `ContainsReason` to me implies like a collection contains method semantics (which is not the case here - as mentioned in the other comment).

Reply:
or `IncludeReason`